File: ebs_flutter/lib/model/ProductCategory.dart
class ProductCategory {
  final int? id;
  final int? productCategoryCode;
  final String? productCategoryName;
  final String? description;

  ProductCategory({
    this.id,
    this.productCategoryCode,
    this.productCategoryName,
    this.description,
  });

  factory ProductCategory.fromJson(Map<String, dynamic> json) {
    return ProductCategory(
      id: json['id'],
      productCategoryCode: json['productCategoryCode'],
      productCategoryName: json['productCategoryName'],
      description: json['description'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'productCategoryCode': productCategoryCode,
      'productCategoryName': productCategoryName,
      'description': description,
    };
  }
}

File: ebs-springboot/src/main/java/com/erp/model/ProductCategory.java
package com.erp.model;

import javax.persistence.*;

@Entity(name = "product_category")
@Table(name = "product_category")
public class ProductCategory {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(name = "product_category_code")
    private Long productCategoryCode;

    @Column(name = "product_category_name")
    private String productCategoryName;

    @Column(name = "description")
    private String description;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getProductCategoryCode() {
        return productCategoryCode;
    }

    public void setProductCategoryCode(Long productCategoryCode) {
        this.productCategoryCode = productCategoryCode;
    }

    public String getProductCategoryName() {
        return productCategoryName;
    }

    public void setProductCategoryName(String name) {
        this.productCategoryName = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }
    
}
